src/main/java/org/apache/sysml/runtime/instructions/cp/BinaryScalarScalarCPInstruction.java

```diff
@@ -65,10 +65,14 @@ else if( so1 instanceof DoubleObject || so2 instanceof DoubleObject || output.ge
 				sores = new DoubleObject( dop.fn.execute(so1.getDoubleValue(), so2.getDoubleValue()) );
 			}
 			else if( so1 instanceof IntObject || so2 instanceof IntObject ) {
```

Comment:
if it is `bitwAnd(1, 2.0)`, then this is being evaluated here itself, but can it be on the next loop.

cc @mboehm7